File: test/Automation_python_CodeGen_Quality_RealRepo_python/pipelines/Python_E2E_Existing_Pipeline_All_1/code/python_e2e_existing_pipeline_all_1/graph/WindowFunction_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from python_e2e_existing_pipeline_all_1.config.ConfigStore import *
from python_e2e_existing_pipeline_all_1.udfs.UDFs import *

def WindowFunction_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn(
          "customer_id",
          row_number()\
            .over(Window\
            .partitionBy(col("customer_id"), col("first_name"))\
            .orderBy(col("phone").asc(), col("email").asc())\
            .rowsBetween(Window.unboundedPreceding, Window.currentRow))
        )\
        .withColumn(
          "country_code",
          row_number()\
            .over(Window\
            .partitionBy(col("customer_id"), col("first_name"))\
            .orderBy(col("phone").asc(), col("email").asc())\
            .rowsBetween(Window.unboundedPreceding, Window.currentRow))
        )\
        .withColumn("account_open_date", row_number()\
        .over(Window\
        .partitionBy(col("customer_id"), col("first_name"))\
        .orderBy(col("phone").asc(), col("email").asc())\
        .rowsBetween(Window.unboundedPreceding, Window.currentRow)))

File: test/Automation_scala_CodeGen_Quality_RealRepo_scala/pipelines/Scala_E2E_Existing_Pipeline_All_1/code/src/main/scala/io/prophecy/pipelines/scala_e2e_existing_pipeline_all_1/graph/WindowFunction_1.scala
package io.prophecy.pipelines.scala_e2e_existing_pipeline_all_1.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_1.udfs.PipelineInitCode._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_1.udfs.UDFs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_1.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object WindowFunction_1 {

  def apply(context: Context, in: DataFrame): DataFrame = {
    import org.apache.spark.sql.expressions.{Window, WindowSpec}
    in.withColumn(
        "c_short",
        row_number().over(
          Window
            .partitionBy(col("c_int"), col("c_long"))
            .orderBy(col("c_float").asc,
                     col("c_string").desc,
                     col("c_string1").asc
            )
            .rowsBetween(Window.unboundedPreceding, Window.currentRow)
        )
      )
      .withColumn(
        "c_boolean",
        row_number().over(
          Window
            .partitionBy(col("c_int"), col("c_long"))
            .orderBy(col("c_float").asc,
                     col("c_string").desc,
                     col("c_string1").asc
            )
            .rowsBetween(Window.unboundedPreceding, Window.currentRow)
        )
      )
  }

}
